test/test-clusters/src/main/java/org/elasticsearch/test/cluster/local/AbstractLocalClusterFactory.java

```diff
@@ -426,7 +430,7 @@ private void writeConfiguration() {
                 try (Stream<Path> configFiles = Files.walk(distributionDir.resolve("config"))) {
                     for (Path file : configFiles.toList()) {
                         Path relativePath = distributionDir.resolve("config").relativize(file);
-                        Path dest = configDir.resolve(relativePath);
+                        Path dest = configDir.resolve(relativePath.toFile().getPath());
```

Comment:
Why do we need to do this path -> file -> path conversion?

Reply:
The last conversion gives a String. The reason is that temporary directory created by Lucene temporariy directory, e.g. `LuceneTestCase#createTempDir` returns a Lucene `FilterDirectory` which always works for `resolve(String)` but throws for `resolve(Path)` if the given `Path` argument is not also a `FilterDirectory`. Since I changed the other PR to use Junit's TemporaryDirectory, this is no longer an issue for now. But I think it might be worthwhile to keep the change so that it does not just break if someone decides to use Lucene temporary in future?

Reply:
I don't have super strong feelings about working around idiosyncrasies with lucene types. Can this be simplified to `relativePath.toString()`?